src/libproc/proc_pid.rs

```diff
@@ -214,10 +214,10 @@ pub fn listpids(proc_types: ProcType) -> Result<Vec<u32>, String> {
         Err(get_errno_with_message(ret))
     } else {
         unsafe {
-            pids = Vec::from_raw_parts(buffer_ptr as *mut u32, ret as usize, pids.capacity());
-            // Seems that the buffer is padded with a lot of pids set to zero
-            pids.retain(|&p| p > 0u32);
+            pids.set_len(ret as usize);
         }
+        // Seems that the buffer is padded with a lot of pids set to zero
+        pids.retain(|&p| p > 0);
```

Comment:
I'm not sure if this `.retain()` call is necessary now -- the extra zeros may have been due to memory unsafety bugs before.